frontend_build/src/read_bundle_plugins.js

```diff
@@ -124,12 +142,27 @@ var readBundlePlugins = function(base_dir) {
 
   mkdirp.sync(locale_dir);
 
+  var nodePaths = [];
+
   // We add some custom configuration options to the bundles that webpack 2 dislikes, clean them up here.
   bundles.forEach(function(bundle) {
     delete bundle.core_name;
     delete bundle.coreAPISpec;
+    if (bundle.nodePaths) {
```

Comment:
These are disallowed yes. We delete them before passing the configuration to webpack, hence no error arises.

We can't use `resolve.modules` because we are using the `nodePaths` to set resolution behaviour for loading webpack loaders and plugins. I gleefully thought that `resolveLoader` would allow this, but unfortunately, because when you set absolute paths in either `resolve.modules` or `resolveLoader` it doesn't treat them the same way as relative paths (where it will do a depth search), so I would either have to list every loader we want accessible, or do this.

Reply:
The `nodePaths` config option looks like it's becoming part of our general plugin API. Do you foresee other uses of it? Or is it more of a one-off hack to get our internal perseus plugin working?

If the latter - at least for now - it might be nice to name the attribute something more scary sounding to dissuade its reuse. E.g. `_perseusInternalNodeModulesPath` :)

Reply:
It's necessary for plugins that need to define their own loaders, which the perseus plugin does. I think I would rather keep it as is for now (rather than futz around with perseus), but it's something that I will hope to resolve more satisfactorily with the build system decoupling project.